zeroforcing/fastqueue.pyx

```diff
@@ -0,0 +1,42 @@
+cdef class FastQueueForBFS:
+    def __init__(self, max_priority):
+        self.array_list = []
+        self.smallest_nonempty_priority = 0
+        self.max_possible_priority = 0
+
+        for i in range(max_priority+1):
+            self.array_list.append(list())
+        self.max_possible_priority = max_priority
+        self.smallest_nonempty_priority = max_priority + 1
+    
+    def __len__(self):
+        cdef int total_length = 0
+        for i in range(self.max_possible_priority+1):
+            total_length += len(self.array_list[i])
+        return total_length
+    
+    cdef object pop(self):
+        cdef int _
+        cdef object popped
+
+        _, popped = self.pop_and_get_priority()
+        return popped
+    
+    cdef tuple pop_and_get_priority(self):
```

Comment:
Could simply store 1st and 2nd highest priority, and adjust as needed when pushes/pops happen.

Reply:
Before change:
```
ncalls   tottime  percall  cumtime  percall filename:lineno(function)
18312381   3.321    0.000    3.321    0.000 zeroforcing/fastqueue.pyx:21(pop_and_get_priority)
 ```

Reply:
After naive change:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 18290879    3.633    0.000    3.633    0.000 zeroforcing/fastqueue.pyx:48(pop_and_get_priority)
 39358815    2.817    0.000    2.817    0.000 zeroforcing/fastqueue.pyx:20(push)
 ```

Reply:
Using dict instead of list of lists:
 ```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 18292738    3.801    0.000    3.801    0.000 zeroforcing/fastqueue.pyx:44(pop_and_get_priority)
 22709554    2.610    0.000    3.530    0.000 bitset_base.pxd:482(__pyx_fuse_0bitset_add)
 39361024    2.841    0.000    2.841    0.000 zeroforcing/fastqueue.pyx:20(push)

Reply:
Updating on push:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 18291572    3.044    0.000    3.044    0.000 zeroforcing/fastqueue.pyx:33(pop_and_get_priority)
 39360546    1.669    0.000    1.669    0.000 zeroforcing/fastqueue.pyx:16(push)
```

Reply:
Holy crap... When using a heapqueue:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     6139    0.001    0.000    0.001    0.000 zeroforcing/fastqueue.pyx:13(push)
      215    0.000    0.000    0.000    0.000 zeroforcing/fastqueue.pyx:30(pop_and_get_priority)
```

The number of calls is drastically smaller which is interesting...
But all 23 graphs in the current test suite pass in **14.55 seconds COMBINED**.

Reply:
Ha! There was a bug. After fixing the bug:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 18295260    3.311    0.000    3.311    0.000 zeroforcing/fastqueue.pyx:30(pop_and_get_priority)
 39365645    2.442    0.000    2.442    0.000 zeroforcing/fastqueue.pyx:13(push)
```